gwpy/timeseries/io/hdf5.py

```diff
@@ -42,6 +42,10 @@ def read_hdf5_timeseries(h5f, path=None, start=None, end=None, **kwargs):
     series = read_hdf5_array(h5f, path=path, **kwargs)
     # crop if needed
     if start is not None or end is not None:
```

Comment:
@duncanmmacleod, is it worth trying to make it so these `if` statements aren't nested? Something like

```python
    if start is not None:
        start = max(start, series.span[0])
    if end is not None:
        end = min(end, series.span[1])
    if start or end:
        return series.crop(start, end)
    return series
```